### DISCUSSION TITLE
Mill 0.12.0-RC1 is out

### DISCUSSION THREAD

**Discussion by lihaoyi** [com-lihaoyi/mill#3497] [News]
> The upcoming 0.12.0 release focuses on improving the scalability of Mill builds: in terms of performance, code organization, adherence to best practices, and community. RC1 is intended to give people a time to try out the major changes so that any issues can be resolved before the release of 0.12.0 final (targeting end-September 2024).

See the changelog for more details https://github.com/com-lihaoyi/mill#0120-rc1---2024-09-09

**Comment by dangdennis** [comment]
Thanks for 0.12! Works fine for my small application so far. However I want to share a small, but resolved, issue. 

After a `mill clean`, using `build.mill` and `0.12.0-RC1`, I discovered that 0.12.0-RC1 generated a new `out/all/compile.super/all.json` unlike the usual `out/all/compile.json`. 

In my `build.sc`, my task `mill app.compile` is a single task to compile all my repo's packages.
```
object all extends CommonModule {
    def compile = T.command {
        api.compile()
        test.compile()
        cli.compile()
    }
}
```

With 0.12.0-RC1 and `build.sc` renamed to `build.mill`, I get error
```
An unexpected error occurred
Exception in thread "MillServerActionRunner" java.nio.file.NoSuchFileException: /Users/dennis/Desktop/projectx/api/out/all/compile.json
```

As shown by the error, `out/all/compile.json` was missing. But we have a new `out/all/compile.super/all.json`. Rewriting the task to below solved my issue, and the `out/all/compile.super/` directory is no longer created.
```
object all extends Module {
  def moduleDeps = Seq(api, test, cli)

  def compile = T {
    T.traverse(moduleDeps)(_.compile)
  }
}
```
